src/main/java/org/helpberkeley/memberdata/HBParser.java

```diff
@@ -606,17 +559,32 @@ static void parseDetails(long postNumber, final String rawPost,
     }
 
     public static String shortURLDiscoursePost(final String line) {
-        int index = line.indexOf(Constants.UPLOAD_URI_PREFIX);
+        UploadFile.auditFilePrefix(line);
+        int index = -1;
+        int prefixLength = 0;
+        if (line.contains(Constants.UPLOAD_URI_PREFIX)) {
+            index = line.indexOf(Constants.UPLOAD_URI_PREFIX);
+            prefixLength= Constants.UPLOAD_URI_PREFIX.length();
+        }
+        else if (line.contains(Constants.WEB_CSV_PREFIX)){
+            index = line.indexOf(Constants.WEB_CSV_PREFIX);
+            prefixLength= Constants.WEB_CSV_PREFIX.length();
+        }
         assert index != -1 : line;
-        String shortURL = line.substring(index);
+        String shortURL = Constants.UPLOAD_URI_PREFIX.concat(line.substring(index + prefixLength));
         index = shortURL.indexOf(')');
         shortURL = shortURL.substring(0, index);
 
         return shortURL;
     }
```

Comment:
This is only used by UploadFile.createUploadFile().  Move this to be a private method in UploadFile.